### PR TITLE
Add filament runout for serial printing

### PR DESCRIPTION
**Pull request by RadekVana**
BFW-1883

### CODE REVIEW

File: src/common/non_file_printing_counter.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * @date 2021-04-29
+ */
+
+#include "non_file_printing_counter.hpp"
+#include "rtos_api.hpp"
+
+unsigned NonFilePrintingCounter::counter = 0;
+
+void NonFilePrintingCounter::increment() {
+    CriticalSection C;
```
Comment (DRracer) [inline src/common/non_file_printing_counter.cpp:13]: Is the critical section really necessary? How about std::atomic? The MCU should have atomic instruction support.

File: src/common/non_file_printing_counter.hpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * @file non_file_printing_counter.hpp
+ * @author Radek Vana
+ * @brief
+ * @date 2021-04-29
+ */
+
+#include <cstdint>
+
+class NonFilePrintingCounter {
+    static unsigned counter;
```
Comment (DRracer) [inline src/common/non_file_printing_counter.hpp:11]: This looks like a singleton, but that probably wasn't the intent. From the usage of this class it looks more like a guard combined with the guard variable into one piece. Looks overengineered to me.

**Review by DRracer — APPROVED** [review]
approving, but there are a few comments worth solving in the future

File: src/common/non_file_printing_counter.hpp (hunk 2 above)
Comment (RadekVana) [reply src/common/non_file_printing_counter.hpp:11]: this is not singleton

**Pull request merged**